_posts/2020-06-18-Metal3-dev-env-BareMetal-Cluster-Deployment.md

```diff
@@ -0,0 +1,396 @@
+---
+title: "Metal³ development environment walkthrough part 2: Deploying a new baremetal cluster"
+draft: false
+categories: ["metal3", "kubernetes", "cluster API", "metal3-dev-env"]
+author: Himanshu Roy
+---
+
+## Introduction
+
+This blog post describes how to deploy a bare metal cluster, a virtual one for simplicity, using [Metal³/metal3-dev-env](https://github.com/metal3-io/metal3-dev-env). We will briefly discuss the steps involved in setting up the cluster as well as some of the customization available. If you want to know more about the architecture of Metal³, this [blogpost]({%post_url 2020-02-27-talk-kubernetes-finland-metal3 %}) can be helpful.
+
+This post builds upon the [detailed metal3-dev-env walkthrough blogpost]({%post_url 2020-02-18-metal3-dev-env-install-deep-dive %}) which describes in detail the steps involved in the environment set up and management cluster configuration. Here we will use that environment to deploy a new Kubernetes cluster using Metal³.
+
+Before we get started, there are a couple of requirements we are expecting to be fulfilled.
+
+<br/>
+
+## Requirements
+
+- Metal³ is already deployed and working,  if not please follow the instructions in the previously mentioned [detailed metal3-dev-env walkthrough blogpost]({%post_url 2020-02-18-metal3-dev-env-install-deep-dive %}).
+- The appropriate environment variables are setup via shell or in the `config_${user}.sh` file, for example -
+  - CAPI_VERSION
+  - NUM_NODES
+  - CLUSTER_NAME
+
+<br/>
+
+
+## Overview of Config and Resource types
+In this section we give a brief overview of the important config files and resources used as part of the BareMetal Cluster deployment. The following sub-sections shows where and which config files, resources are created and give a brief description about some of them. These will help you understand the technical details of the cluster deployment. You can also skip this section, visit the next section about *provisioning* first and then revisit this 
+
+
+### Config Files and Resources Types
+
+!["A list of files under manifests directory for CLUSTER_NAME=bmetalcluster"](/assets/2020-06-18-Metal3-dev-env-BareMetal-Cluster-Deployment/manifest-directory.png)  
+> info "Note"
+> All these files are generated under the path `https://github.com/metal3-io/metal3-dev-env/tree/master/vm-setup/roles/v1aX_integration_test/files` as part of the provisioning process. When you do a provisioning for the first time, this directory is created and is subsequently used to store the templates and config files generated for deploying the baremetal cluster.
+
+<br/>
```

Comment:
I don't know a lot about the tools used to publish the site. Are the `<br/>` tags in the file important to the blog post parser?

Reply:
There was an issue with the next heading not having much gap with previous section causing the blog to look cramped and difficult to read. I had to force line breaks using `<br/>` tags to fix that.